File: Easy/1. Two Sum/solution.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        ans = []
        for i, num1 in enumerate(nums):
            for j, num2 in enumerate(nums):
                if i == j:
                    continue
                if num1 + num2 == target:
                    ans.append(i)
                    ans.append(j)
                    break
            if ans != []:
                break
        return ans

# Time complexity: O(n^2)
# Space complexity: O(1)

# another solution

File: Easy/1. Two Sum/solution.java
class Solution {
  public int[] twoSum(int[] nums, int target) {
      Map<Integer, Integer>numToIndex = new HashMap<>();

      for (int i=0; i < nums.length; i++){
          if (numToIndex.containsKey(target - nums[i])){
              int[] ans = {numToIndex.get(target-nums[i]), i};
              return ans;
          } else {
              numToIndex.put(nums[i], i);
          }
      }
      return null;
  }
}

// Time Complexity: O(n)
// Space Complexity: O(n)